File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
       return new int[]{0,2,5,3,7,8,4};
    }

    public int getToday() {
        return birdsPerDay[birdsPerDay.length-1];
    }

    public void incrementTodaysCount() {
        birdsPerDay[birdsPerDay.length-1]++;
    }

    public boolean hasDayWithoutBirds() {
        boolean birds = false;
        for(int bird: birdsPerDay){
            if(bird==0){
                return true;
            }
        }
        return birds;
    }

    public int getCountForFirstDays(int numberOfDays) {
        int countBirds=0;
        if(numberOfDays < birdsPerDay.length){
            for (int i=0; i<=birdsPerDay[numberOfDays-1]; i++){
                countBirds= countBirds+birdsPerDay[i];
            }
        }else{
            for (int i=0; i<birdsPerDay.length; i++){
                countBirds= countBirds+birdsPerDay[i];
            }
        }
        return countBirds;
    }

    public int getBusyDays() {
        int busyDays=0;
        for (int i=0; i<birdsPerDay.length; i++){
            if(birdsPerDay[i] >= 5){
                busyDays++;
            }
        }
    return busyDays;
    }
}

File: solutions/csharp/bird-watcher/1/BirdWatcher.cs
class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek()
    {
        return new int[] {0,2,5,3,7,8,4};
    }

    public int Today()
    {
        return birdsPerDay.Last();
    }

    public void IncrementTodaysCount()
    {
        birdsPerDay[6]++;
    }

    public bool HasDayWithoutBirds()
    {
        bool noBirds = false;
        foreach (int bird in birdsPerDay){
            if (bird == 0){
                noBirds = true;
            }
        }
        return noBirds;
    }

    public int CountForFirstDays(int numberOfDays)
    {
        int count = 0;
        for(int i=0; i<numberOfDays; i++){
            count += birdsPerDay[i];
        }
        return count;
    }

    public int BusyDays()
    {
        int countBusyDays = 0;
        foreach (int bird in birdsPerDay){
            if (bird >= 5){
                countBusyDays++;
            }
        }
        return countBusyDays;    
    }
}
